### DISCUSSION TITLE
✨Improve Ubuntu CLI packaging (aka Installer script for bash completion leads to yak shave)

### DISCUSSION THREAD

**Discussion by blaisep** [dagger/dagger#12488] [Ideas]
### What are you trying to do?

Installing dagger-cli from Homebrew is awesome.
less awesome is installing by
```
cd /usr/local
curl -L https://dl.dagger.io/dagger/install.sh | sh
```

I searched for "package" "install" and "completion" before opening this issue.
Labels: CLI, DX
(There is a thread describing the situation in detail at https://discord.com/channels/707636530424053791/1155601074393194658/1155601074393194658 )

### Why is this important to you?

I think that it will be easier for new users to adopt dagger if the onboarding is familiar and safe. 
For example, Homebrew is great, but is less common on Linux.

### How are you currently working around this?

I decided to `sudo curl` which is generally not a good idea.

Bash completions are more complicated because things like bash completions require special knowledge of how bash works on the local machine, etc. Again specifics are in the Discord thread and I'm happy to move it over here.

**Comment by TomChv** [comment]
@gerhard This issue is related to our release process and avaibility, I let you be in charge when you have bandwidth

**Comment by gerhard** [comment]
Thanks for opening this @blaisep!

The script can be used to install the binary anywhere, it doesn't have to be `/usr/local`.

This is how I use it:

```sh
curl https://dl.dagger.io/dagger/install.sh | BIN_DIR=~/.local/bin sh
# ...
sh info installed /Users/gerhard/.local/bin/dagger
```

Since `~/.local/bin` is in my `$PATH`, this is what I end up with:

```sh
which dagger
/Users/gerhard/.local/bin/dagger

dagger version
dagger v0.9.0 (registry.dagger.io/engine) darwin/amd64
```

Having read https://discord.com/channels/707636530424053791/1155601074393194658/1155601074393194658, I think that you would be more comfortable if the script installed to `~/.dagger/bin` by default.

It would also make sense for the shell completions to be installed in `~/.dagger/completions`

WDYT @blaisep ?

**Comment by blaisep** [comment]
Thank you @gerhard

**Comment by gerhard** [comment]
I stumbled across this today and reminded myself that there are 2 action items here:
1. Install to `~/.dagger/bin` by default
2. Install shell completions to `~/.dagger/completions` by default

Took a note to come back to this when I'm finished with the current area of focus. Marking it as open so that it signals to others that this still needs addressing.

Thanks @blaisep 💪

**Comment by blaisep** [comment]
...and this --^ is what makes @gerhard a rock star.
